packages/typescript/src/lib/TypescriptCompiler.ts

```diff
@@ -112,26 +101,44 @@ export class TypescriptCompiler {
       sourceFiles,
       data
     ) => {
-      // for exposes: { "./expose/path": "path/to/file" }
-      // force typescript to write compiled output to "@mf-typescript/expose/path"
-      const sourceFilename = sourceFiles?.[0].fileName || '';
-
-      // Try to rewrite the path with exposed federated modules,
-      // failing so, use the default filepath emitted by TS Compiler.
-      // This second case is valid for 'additionalFileToCompiler' added through Plugin Options.
-      const normalizedFilepath =
-        rewritePathsWithExposedFederatedModules(sourceFilename) ?? filepath;
-
-      this.tsDefinitionFilesObj[normalizedFilepath] = text;
-
+      this.tsDefinitionFilesObj[filepath] = text;
       originalWriteFile(
-        normalizedFilepath,
+        filepath,
         text,
         writeOrderByteMark,
         onError,
         sourceFiles,
         data
       );
+
+      // create exports matching the `exposes` config
+      const sourceFilename = sourceFiles?.[0].fileName || '';
+      const exposedDestFilePath = exposeSrcToDestMap[sourceFilename];
+
+      // create reexport file only if the file was marked for exposing
+      if (exposedDestFilePath) {
+        const normalizedExposedDestFilePath = path.join(
+          this.options.distDir,
+          `${exposedDestFilePath}.d.ts`
+        );
+
+        const relativePathToCompiledFile =
+          './' +
+          path
+            .relative(
+              path.join(normalizedExposedDestFilePath, '../'), // import relative to the folder which contains this file
+              filepath
+            )
+            .replace(/\.d\.ts$/, '');
+
+        const reexport = `export * from '${relativePathToCompiledFile}';\nexport { default } from '${relativePathToCompiledFile}';`;
```

Comment:
```suggestion
        const relativePathToCompiledFile =
          path
            .relative(
              path.dirname(normalizedExposedDestFilePath), // import relative to the folder which contains this file
              filepath
            ).replace(/\.d\.ts$/, '');

        const reexport = `export * from './${relativePathToCompiledFile}';\nexport { default } from '${relativePathToCompiledFile}';`;
```

Also, maybe perhaps you can refactor this a little for much more readability?

Reply:
Yeah, your example was a lot cleaner, I will address this later